plugins/tech-insights-backend-module-jsonfc/src/service/JsonRulesEngineFactChecker.ts

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { JsonRuleBooleanCheckResult, TechInsightJsonRuleCheck } from '../types';
+import {
+  FactChecker,
+  FactResponse,
+  FactValueDefinitions,
+  TechInsightFact,
+  TechInsightsStore,
+} from '@backstage/plugin-tech-insights-common';
+import { Engine, EngineResult, TopLevelCondition } from 'json-rules-engine';
+import {
+  DefaultCheckRegistry,
+  TechInsightCheckRegistry,
+} from './CheckRegistry';
+import { Logger } from 'winston';
+import { pick } from 'lodash';
+
+type JsonRulesEngineFactCheckerOptions = {
+  checks: TechInsightJsonRuleCheck[];
+  repository: TechInsightsStore;
+  logger: Logger;
+  checkRegistry?: TechInsightCheckRegistry<any>;
+};
+
+const noopEvent = {
+  type: 'noop',
+};
+
+class JsonRulesEngineFactChecker
+  implements FactChecker<TechInsightJsonRuleCheck, JsonRuleBooleanCheckResult>
+{
+  private readonly checkRegistry: TechInsightCheckRegistry<TechInsightJsonRuleCheck>;
+  private repository: TechInsightsStore;
+  private readonly logger: Logger;
+
+  constructor({
+    checks,
+    repository,
+    logger,
+    checkRegistry,
+  }: JsonRulesEngineFactCheckerOptions) {
+    this.repository = repository;
+    this.logger = logger;
+    checks.forEach(check => this.validate(check));
+    this.checkRegistry = checkRegistry ?? new DefaultCheckRegistry(checks);
+  }
+
+  async runChecks(
+    entity: string,
+    checks: string[],
+  ): Promise<JsonRuleBooleanCheckResult[]> {
+    const engine = new Engine();
+    const techInsightChecks = this.checkRegistry.getAll(checks);
+    const factRefs = techInsightChecks.flatMap(it => it.factRefs);
```

Comment:
Super trivial but I'm not sure about the semantics of "ref" (technically it's an abbreviation). It's just not all that clear what it means (though it's clear enough).

Reply:
@iain-b Any ideas on alternative nomenclature?